aiogram/contrib/middlewares/fsm.py

```diff
@@ -63,18 +57,18 @@ def state(self):
         self._is_dirty = True
 
     async def save(self, force=False):
-        if self._copy != self or force:
+        if self._copy != dict(self) or force:
             await self.fsm_context.set_data(data=self)
         if self._is_dirty or force:
             await self.fsm_context.set_state(self.state)
         self._is_dirty = False
-        self._copy = copy.deepcopy(self)
+        self._copy = copy.deepcopy(dict(self))
```

Comment:
do a `dict(self)` once, lint a name to it and use it everywhere, where you use `self` as data

Reply:
self._copy is used to not to save changes when there isn't.
therefore dict(self) must be used every time whed we need to compare self as data with his copy.